iceoryx_posh/include/iceoryx_posh/popo/modern_api/typed_subscriber.hpp

```diff
@@ -42,7 +42,8 @@ class TypedSubscriber : public base_subscriber_t
     void unsubscribe() noexcept;
 
     bool hasNewSamples() const noexcept;
-    cxx::expected<cxx::optional<Sample<const T>>, ChunkReceiveError> receive() noexcept;
+    bool hasMissedSamples() noexcept;
+    cxx::expected<cxx::optional<Sample<const T>>, ChunkReceiveError> take() noexcept;
```

Comment:
This expected of optional is supposed to change in the future right? Since it is a little cumbersome to use. Maybe as a specific sample type combining the two...not sure what would be the best approach.

Reply:
See https://github.com/eclipse/iceoryx/issues/340#issuecomment-724149533 
Feedback appreciated!